services/core-web/src/components/Forms/GenerateDocumentForm.js

```diff
@@ -104,6 +107,16 @@ export const GenerateDocumentForm = (props) => {
             Cancel
           </Button>
         </Popconfirm>
+        <Button
```

Comment:
I think we should return the promise from the onClick props.preview and set an "isGeneratingPreview" flag in the state here to set the loading prop of this button to prevent users from spam clicking it. Buttons should not be spammable while they are already performing the initial request.

Reply:
I pulled it out into a handle preview function and added some state to handle disabling the button.